README.md

```diff
@@ -1,11 +1,56 @@
-# multiphase-stokes
-Contains an operator class for variable-coefficient two-phase staggered stokes. See *VCTwoFluidStaggeredStokesOperator.cpp*.
+# Adaptive Mesh Refinement for Multiphase Fluids
 
-There are two driver programs:
-+ **apply.cpp**: Used to test that the operator is correct. This code applies the operator to user-defined network velocity, solvent velocity and pressure. The result is checked against the analytical solution to ensure the operator is 2nd order accurate.
-+ **solver.cpp**: This solves the coupled two-phase system using a krylov method (GMRES). The krylov method requires the operator to be passed to it. 
-+ **smooth.cpp**: This uses the stand-alone box relaxation scheme with gauss-seidel type iterations to solve the fluid equations
+This repository contains the source code to simulate multiphase flows on adaptively refined grids. The solver implemented here is second order accuracte in $L^1$, $L^2$, $L^\infty$ norms for all solution variables of the Newtonian and non-Newtonian models. Our experiments demonstrate the solver is stable provided $\Delta t$ satisfies the imposed CFL condition. The solver can accurately resolve sharp gradients in the solution and, with the multigrid preconditioner, its behavior is independent of grid spacing. This AMR solver offers a major cost savings benefits by using refined grids in areas of interest and using coarser grids elsewhere.
 
-## FAC Precondioner Stragegy: VCTwoFluidStaggeredStokesBoxRelaxationFACOperator.cpp
+# Dependencies
 
-Contains routines for smoothening the error using box relaxation, computing residual, prolongation, restriction and solving on coarsest level.
+You will need to [build and install IBAMR](https://ibamr.github.io/linux) and all its required dependencies. We recommend compiling the optimized builds of the dependencies if you simply want to try out some simulations. Additionally, you should compile IBAMR with CMake since we provide a CMakeLists.txt to compile the source code in this repository. 
+
+The simulations generated can be viewed by installing [VISit](https://visit-dav.github.io/visit-website/) on your machine.
+
+# Compiling the code
+
+1. In the project’s root directory, create a build directory so that all object files and executables are stored separately from the source code.
+
+```
+mkdir build
+```
+
+2. Enter the build directory and run cmake to configure and generate build files for our `multiphaseLib` library, and link it to IBAMR. The command below uses the `O3` compiler optimization flag to enable level 3 optimizations. 
+
+
+```
+cd build
+cmake /path/to/CMakeLists.txt \
+    -DIBAMR_ROOT=/path/to/ibamr/install/dir \
+    -DCMAKE_CXX_FLAGS="-O3"
+```
+Note: If cmake fails, then make sure you're using the same compilers that were used for compiling IBAMR. You may provide a path to your compilers by passing flags for `-DCMAKE_CXX_COMPILER`, `-DCMAKE_C_COMPILER` and `-DCMAKE_Fortran_COMPILER`.
```

Comment:
We do not require the same compilers. It should work as long as the compilers are ABI consistent.
```suggestion
```